www/js/draw-model.js

```diff
@@ -0,0 +1,155 @@
+			this._model = model;
+			var vertices = [];
+			var temp = []
+			vertices = array.map(model.getNodes(), function(node){
+				temp = temp.concat(this.addNode(node));
+				return temp;
+			}, this);
+			vertices = vertices[vertices.length - 1]; //hack for keeping it one dimension
+			console.log(vertices);
+
+			instance.doWhileSuspended(function(){
+
+				array.forEach(vertices, function(vertex){
+					instance.makeSource(vertex, {
+						filter: ".ep",
+						anchor: "Continuous",
+						connector: ["StateMachine", {curviness: 0}],
+						connectorStyle:{ strokeStyle:"#5c96bc", lineWidth:2, outlineColor:"transparent", outlineWidth:4 },
+						maxConnections: 6,
+						onMaxConnections: function(info, e){
+							alert("Maximum connections (" + info.maxConnections + ") reached");
+						},
+					});
+				});
+
+				array.forEach(vertices, function(vertex){
+					instance.makeTarget(vertex, {
+						dropOptions: {hoverClass: "dragHover"},
+						anchor: "Continuous"
+					});
+				});
+			});
+
+			array.forEach(vertices, function(vertex){
+				var id = attr.get(vertex, "id");
+				var links = model.getLinks(id);
+				//unlike Dragoon everyone does not have links.
+				//only equation nodes have the links.
+				if(links)
+					this.setConnections(links, vertex);
+			}, this);
+
+			return instance;
+		},
+
+		addNode: function(/* object */ node){
+			type = node.type || "circle";
+			console.log("Adding vertex to the canvas id = ", node.ID, " type = ", type);
+
+			console.log("Position for the vertex : ",node.ID, " position: x ", node.position[0].x, " y: " + node.position[0].y);
+			var properties = this.getNodeUIProperties(node);
+			var htmlStrings = graphObjects.getNodeHTML(this._model, node.ID); 
+
+			var vertices = [];
+			array.forEach(htmlStrings, function(html, count){
+				var idTag = node.ID;
+				if(count == 1)
+					idTag += "_initial"
+				vertices[count] = domConstruct.create("div", {
+					id: idTag,
+					"class": type,
+					style: {
+						left: node.position[count].x + 'px',
+						top: node.position[count].y + 'px',
+						backgroundColor: properties.backgroundColor,
+						borderColor: properties.borderColor
+					},
+					innerHTML: htmlStrings[count]
+				}, "statemachine-demo");
+			}, this);
+
+			return vertices;
+		},
+
+		getNodeUIProperties: function(node){
+			// node properties default case
+			var obj = {
+				backgroundColor: "DarkGray",
+				borderColor: "black"
+			};
+			var type = node.type;
+			if(type && type == "equation"){
+				obj.backgroundColor = this.getNextColor(true);
+				obj.borderColor = "black";
+			} else {
+				obj.backgroundColor = "white";
+				obj.borderColor = this.getNextColor(false);
+			}
+
+			return obj;
+		},
+
+		getNextColor: function(isBackground){
+			var index = isBackground ? this._backgroundColor++ : this._borderColor--;
+			if(isBackground && index >= this._colors.length){
+				console.error("need more colors, last color returned");
+				index--;
```

Comment:
Yeah this will highlight we need to add more colors. Ok I will just set it to loop it through.